### PR TITLE
EnsureOakIndex unittests

### PR DESCRIPTION
**Pull request by joerghoh**
and finally some refactoring and unittests

### CODE REVIEW

**Comment by joerghoh** [comment]
Travis claims, that the build fails at some test cases, but I cannot reproduce it locally (OSX).

File: bundle/src/main/java/com/adobe/acs/commons/oak/impl/EnsureOakIndexManagerImpl.java [hunk 1]
```diff
@@ -74,7 +74,7 @@
 @Service(value = { DynamicMBean.class, EnsureOakIndexManager.class })
 public class EnsureOakIndexManagerImpl extends AnnotatedStandardMBean implements EnsureOakIndexManager, EnsureOakIndexManagerMBean {
 
-    private static final Logger log = LoggerFactory.getLogger(EnsureOakIndexManagerImpl.class);
+    private static final Logger LOG = LoggerFactory.getLogger(EnsureOakIndexManagerImpl.class);
```
Comment (davidjgonzalez) [inline bundle/src/main/java/com/adobe/acs/commons/oak/impl/EnsureOakIndexManagerImpl.java]: Any reason why changing to LOG? Project convention is lowercase

Comment (joerghoh) [inline bundle/src/main/java/com/adobe/acs/commons/oak/impl/EnsureOakIndexManagerImpl.java]: IIRC PMD complained about it.

Comment (justinedelson) [inline bundle/src/main/java/com/adobe/acs/commons/oak/impl/EnsureOakIndexManagerImpl.java]: PMD complained about final not being there

**Pull request merged**